SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp

```diff
@@ -204,24 +215,38 @@ bool ValidateIdentifierNames(TArray<TSharedPtr<FUnrealType>>& TypeInfos)
 		UClass* Class = Cast<UClass>(TypeInfo->Type);
 		check(Class);
 		const FString& ClassName = Class->GetName();
+		const FString& ClassPath = Class->GetPathName();
 		FString SchemaName = UnrealNameToSchemaName(ClassName);
 
-		if (!CheckSchemaNameValidity(SchemaName, Class->GetPathName(), TEXT("Class")))
+		if (!CheckSchemaNameValidity(SchemaName, ClassPath, TEXT("Class")))
 		{
 			bSuccess = false;
 		}
 
+		AddNameCollision(SchemaName, ClassPath);
```

Comment:
Edge case, not really that important:
`(Unreal)A` -> `(schema)A`
`(Unreal)A!` -> collides with `(schema)A` -> `(schema)A1`
`(Unreal)A1` -> collides with `(schema)A1` -> `(schema)A11`
Here, `(Unreal)A!` won't be added in `AddNameCollision` for `(schema)A1`, only for `(schema)A`, so it won't log that `(Unreal)A1` had a collision.